stdlib/public/core/StringStorage.swift

```diff
@@ -390,6 +401,34 @@ extension __StringStorage {
     return __StringStorage.create(
       realCodeUnitCapacity: realCapacity, countAndFlags: countAndFlags)
   }
+  
+  @_effects(releasenone)
+  @usableFromInline
+  internal static func create(
+    unsafeUninitializedCapacity capacity: Int,
+    initializingRepairingUTF8With initializer: (
+      _ buffer: UnsafeMutableBufferPointer<UInt8>,
+      _ initializedCount: inout Int
+    ) throws -> Void
+  ) rethrows -> __StringStorage {
+    let storage = __StringStorage.create(
+      capacity: capacity,
+      countAndFlags: CountAndFlags(mortalCount: 0, isASCII: false)
+    )
+    let buffer = UnsafeMutableBufferPointer(start: storage.mutableStart,
+                                            count: capacity)
+    var count = 0
+    try initializer(buffer, &count)
+    if case .success(let info) =
+      validateUTF8(UnsafeBufferPointer(buffer)) {
+      storage._countAndFlags = CountAndFlags(mortalCount: count,
+                                            isASCII: info.isASCII)
+      storage._invariantCheck()
+      return storage
+    } else {
+      return String._fromUTF8Repairing(UnsafeBufferPointer(buffer)).result._guts._object.nativeStorage
```

Comment:
This is obviously a ludicrous way to accomplish this. I'll find something better before landing it.